src/App.tsx

```diff
@@ -1,14 +1,58 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState('All');
+  const [query, setQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [showModal, setShowModal] = useState(false);
+  const [todoModalId, setTodoModalId] = useState(0);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getTodos()
+      .then(todosList => {
+        const todoFilter = todosList
+          .filter(todo => todo.title
+            .toLowerCase()
+            .includes(query.trim().toLowerCase()))
+          .filter(todo => {
+            switch (selectedFilter) {
+              case 'active':
```

Comment:
Let's create a separate function for filtering. And do you really need to call `filter` method twice?